src/allmydata/immutable/layout.py

```diff
@@ -195,6 +195,14 @@ def put_block(self, segmentnum, data):
         return self._write(offset, data)
 
     def put_crypttext_hashes(self, hashes):
+        # plaintext_hash_tree precedes crypttext_hash_tree. It is not used, and
+        # so is not explicitly written, but we need to write everything, so
+        # fill it in with nulls.
+        d = self._write(self._offsets['plaintext_hash_tree'], b"\x00" * self._segment_hash_size)
+        d.addCallback(lambda _: self._really_put_crypttext_hashes(hashes))
+        return d
```

Comment:
Previously, over Foolscap ... the client just jumped over these bytes, never writing them?  Oof.

Reply:
Everywhere it's mentioned, plaintext-hash-tree says it is unused (I assume "historical artifact"?)

Reply:
1. Previously Foolscap just relied on the fact that the writing API would create null bytes, yeah.
2. Probably historical artifact, yeah, but it's still part of the format so can't just remove it.